pkg/conversion/conversion.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conversion
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/checker/decls"
+	"google.golang.org/genproto/googleapis/api/expr/v1alpha1"
+
+	"k8s.io/apiextensions-apiserver/pkg/apiserver/schema"
+	crdcel "k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel"
+	"k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/library"
+	"k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model"
+
+	"github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+)
+
+const celCheckFrequency = 100
+
+// CompiledRule contains the compiled cel.Program to convert a single field. If there was a compilation error, that is
```

Comment:
```suggestion
// CompiledRule contains the compiled cel.Program to convert a single field. If there were a compilation error, that would be
```

(if we care about correct mood)